service/chartconfig/v1/helm/helm.go

```diff
@@ -46,21 +47,30 @@ func New(config Config) (*Client, error) {
 	return newHelm, nil
 }
 
-// GetReleaseContent gets the current status of the Helm Release.
+// GetReleaseContent gets the current status of the Helm Release including any
+// values provided when the chart was installed.
 func (c *Client) GetReleaseContent(customObject v1alpha1.ChartConfig) (*Release, error) {
 	releaseName := key.ReleaseName(customObject)
 
 	resp, err := c.helmClient.ReleaseContent(releaseName)
-	if IsReleaseNotFound(err) {
+	if err != nil && IsReleaseNotFound(err) {
 		return nil, microerror.Maskf(releaseNotFoundError, releaseName)
 	}
 	if err != nil {
 		return nil, microerror.Mask(err)
 	}
 
+	// Raw values are returned by the Tiller API and we convert these to a map.
+	raw := []byte(resp.Release.Config.Raw)
+	values, err := chartutil.ReadValues(raw)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
 	release := &Release{
 		Name:   resp.Release.Name,
 		Status: resp.Release.Info.Status.Code.String(),
+		Values: values.AsMap(),
```

Comment:
`AsMap` returns the structure we need.

Reply:
Just for clearity: `values` are the values provided by the helm client overriding the charts default `values.yaml`?

Reply:
Yes it just returns the set values which is good for our use-case. We'll need the values in the CRD and will reconcile them to update the installed chart when they change.